addons/udes_sale_stock/models/sale_order.py

```diff
@@ -299,11 +299,15 @@ def action_confirm(self):
         """ Override to disable tracking by default """
         return super(SaleOrder, self.with_context(tracking_disable=True)).action_confirm()
 
+    def _get_confirmation_domain(self):
+        """Returns the domain for sale orders to attempt confirmation."""
+        return [("state", "=", "draft")]
+
     def confirm_orders(self):
         """Process sale and cancel, confirm or ignore sale orders according to
```

Comment:
It is not from your current changes but this docstring is not very helpful.

Reply:
Updated